core/src/main/java/hudson/model/AbstractBuild.java

```diff
@@ -880,15 +880,12 @@ public EnvVars getEnvironment(TaskListener log) throws IOException, InterruptedE
         if (ws!=null)   // if this is done very early on in the build, workspace may not be decided yet. see HUDSON-3997
             env.put("WORKSPACE", ws.getRemote());
 
-        project.getScm().buildEnvVars(this,env);
+        project.getScm().buildEnvironment(this, env);
 
         if (buildEnvironments!=null)
             for (Environment e : buildEnvironments)
                 e.buildEnvVars(env);
 
-        for (EnvironmentContributingAction a : getActions(EnvironmentContributingAction.class))
-            a.buildEnvVars(this,env,getBuiltOn());
```

Comment:
Unnecessary since `super` calls it.

Reply:
Well turns out `EnvironmentVariableNodePropertyTest.testSlaveAndMasterPropertyAndParameterOnSlave` specifically requests [this order](https://github.com/jenkinsci/jenkins/blob/8fa3c39daa84bed96ecaea1fb636335096e303d0/test/src/test/java/hudson/slaves/EnvironmentVariableNodePropertyTest.java#L71). Why, I am not sure, but easier to just be on the safe side.